### ISSUE TITLE
Install from source errors

### ISSUE THREAD

**Issue by stefanve** [wwmm/pulseeffects#94]
I guess Im doing something wrong, but I've followed the instructions at
https://github.com/wwmm/pulseeffects/wiki/Installation-from-Source

Dependiecies install works until:
E: Unable to locate package libstreamer-plugins-bad1.0-dev
After removing that plugin from what copied from install page, it installed all the other modules

Yesterday I had Ubuntu 17.04 and the error was, don't remember exactly but, "wrong parameters to Project(), minimum project name and one language is required"

After upgrading to Ubuntu 17.10 i get:
Meson encountered an error in file meson.build, line 12, column 0:
Native dependency 'libpulse' not found

I cannot find that module either

**Comment by wwmm** [comment]
It must be a typo in the wiki. This package name is probably `libgstreamer-plugins-bad1.0-dev`

The `wrong parameters to Project(), minimum project name and one language is required` may be related to a too old version of meson.

At least in Arch Linux there is a package named `libpulse` that install the file `/usr/lib/pkgconfig/libpulse.pc`. The last error is happening because pkg-config could not find it

**Comment by stefanve** [comment]
OK, with those adjustments it installed. For Ubuntu it is `libpulse-dev`
Not sure it is working right though, no sound is coming through when try to run apps through it.
Seems like a good thing, but maybe not for me. If I run for instance Spotify there's no sound until I toggle it off. When On I see the spectrum, can adjust the EQ and so on, but no sound out.

**Comment by wwmm** [comment]
Can you post a snapshot of pavucontrol window showing the `playback`, `recording` and `output devices` tab while PulseEffects is running and there is no sound? Some time ago a few users had a similar problem and there was not much I could do from my side. For some reason on a few machines gstreamer is not obeying the `set output device` instruction. With the snapshots I can tell if it is the same problem

**Comment by AsavarTzeth** [comment]
`wrong parameters to Project(), minimum project name and one language is required` means your version of Meson is too old. Older versions require you to specify a compiler language. Later versions allow you to skip this, which was implemented to allow languages that don' need this, like python.

Normally Meson is supposed to tell you it is too old. However it can't because the version required is specified under `Project()`.

**Comment by loxaxs** [comment]
`libpulse-dev` should be added to the list of dependencies of Ubuntu 17.10 / Debian testing.
___
I installed pulseeffects on Ubuntu 17.10 artful today. I didn't know what to do when the error `E: Unable to locate package libstreamer-plugins-bad1.0-dev` showed up. I lost a lot of time trying to work my way around it by fixing v3.0.0. I couldn't have install it without this thread.

Thus, thank you stefanve for posting the problem.

Also, thank you wwmm, for that great project. That equalizer tool was really missing on Linux. It does not feel completely mature yet, but it is very useful.

**Comment by wwmm** [comment]
Thanks! I updated the wiki.

PulseEffects will keep evolving. There is still things to do. Specially when it comes to calibration.

**Comment by stefanve** [comment]
Sorry I did not come back to this. I tried it again and now it worked, seems the problem was that when pulseeffects starts it creates new sound devices and selects them, in my case “pulseffects mic”. No sound come from there, regardless of my settings. Then, first time, I believe I picked the wrong “real” output, the digital out from my amplifier rather than the real SPDIF at the computer. Therefore I had a sound, much lower in volume than it should, but it was there as long as pulseeffcts are on bypass.

Long story short: when I select the audio device that was selected before pulseffects change it, everything works fine!

It is a great thing, hope you continue to develop it.

Thanks

**Comment by stefanve** [comment]
I think the fact that the application automatically selects an unusable device can be considered a bug. Specially since there’s no indication that device changed and no indication of which device is being used.

**Comment by wwmm** [comment]
I hope to be able to let the user choose a device someday. Right now what PulseEffects does is asking Pulseaudio which device is set as default and then uses it to play audio. If you run PulseEffects from command line it will print there which device Pulseaudio is telling to be default.

In order to apply effects to an application output a null sink is created (PulseEffects_apps) and the application is redirected to send audio to this sink. The gstreamer pipeline running in PulseEffects "records" from this null sink monitor, applies the effects and then plays the result to the default Pulseaudio output device. 

The null sink Pulseeffects_mic is used when applying effects for microphones. A second gstreamer pipeline records from the mic, applies effects and then plays to this null sink. Recording applications the record from this null sink monitor will capture the mic sound with effects applied. PulseEffects_mic is not used when you are only applying effects to applications like spotify or chrome.
 
What I think that happened is that the device set as default(fallback in pavucontrol window) output in Pulseaudio was not the one you were expecting to have the audio coming through. Or this default output was changed when PulseEffects window was already open. We do not change output device on the fly yet. We detect them only when PulseEffects starts. This will improve in the future.
